### PR TITLE
Include osm3sMeta data in received objects; add to GeoJSON(); emit 'load' at every successful request

### PR DESCRIPTION
**Pull request by plepe**

### CODE REVIEW

File: src/OverpassFrontend.js [hunk 1]
```diff
@@ -249,6 +249,13 @@ class OverpassFrontend {
       this.errorCount = 0
     }
 
+    let osm3sMeta = {
+      osm3s_version: results.version,
+      generator: results.generator,
+      osm_timestamp: results.osm3s.timestamp_osm_base,
```
Comment (promofu) [inline src/OverpassFrontend.js]: here there might as well be some timestamp for "area" extension available if you include the "area" as BBOX... don't have a sample handy but could produce one

Comment (promofu) [reply src/OverpassFrontend.js]: but i like these changes. they would surely make a better solution given the subrequest stuff

Comment (promofu) [reply src/OverpassFrontend.js]: As an example for generator-side extension metadata you might want to try this query in overpass-turbo:

```
[out=json];
( area[name="Wien"]; )->.searchArea;
  node
    [amenity="drinking_water"]
    (area.searchArea);
  out meta;
```

it utilizes the "area" extension of osm3s and results in this

```
{
  "version": 0.6,
  "generator": "Overpass API 0.7.55.4 3079d8ea",
  "osm3s": {
    "timestamp_osm_base": "2018-08-31T20:32:02Z",
    "timestamp_areas_base": "2018-08-31T19:27:02Z",
    "copyright": "The data included in this document is from www.openstreetmap.org. The data is made available under ODbL."
  },
  "elements": [
  ...
```

note the `timestamp_areas_base` key of `osm3s` whereas a "simple" bounding box query would not produce that key as the extension is not involved.

I don't know how many and which extensions are available - and overpass-frontend should not assume the existence of one or any of them beyond maybe osm_base as these are enabled by the resulting query itself.
Thus I would prefer some transparent implementation.

**Comment by plepe** [comment]
Sorry, I haven't reacted on your comments yet. What do you think of the last commit? The data is flattend, but copied as it is (and will therefore include any extensions). Result would look like this:
```json
{
  "version": 0.6,
  "generator": "Overpass API 0.7.55.4 3079d8ea",
  "timestamp_osm_base": "2018-09-28T11:58:02Z",
  "copyright": "The data included in this document is from www.openstreetmap.org. The data is made available under ODbL."
}
```

**Comment by plepe** [comment]
Oh no, that would clash with the 'version' of the object in GeoJSON output.

**Comment by plepe** [comment]
I prefixed osm3s metadata with '@osm3s:'.

So that's how GeoJSON() output now looks like:
```json
{
  "type": "Feature",
  "id": "node/1394233340",
  "properties": {
    "@id": "node/1394233340",
    "amenity": "fast_food",
    "cuisine": "Cornish",
    "name": "West Cornwall Pasty Company",
    "@timestamp": "2011-08-10T22:29:56Z",
    "@version": 1,
    "@changeset": 8981310,
    "@user": "Chris Parker",
    "@uid": 51722,
    "@osm3s:version": 0.6,
    "@osm3s:generator": "Overpass API 0.7.55.4 3079d8ea",
    "@osm3s:timestamp_osm_base": "2018-09-28T12:05:02Z",
    "@osm3s:copyright": "The data included in this document is from www.openstreetmap.org. The data is made available under ODbL."
  },
  "geometry": {
    "type": "Point",
    "coordinates": [
      -0.0861372,
      51.5049528
    ]
  }
}
```

Do you think, that's feasible?

**Comment by plepe** [comment]
So, now OverpassFrontend will also emit a 'load' event, every time data is received from the  server. I think I will merge soon.

**Comment by promofu** [comment]
this time it was me - basically because of vacation... but now I will take a look ;-)

**Comment by promofu** [comment]
and i like it ;-)

**Comment by plepe** [comment]
You are welcome :-)

**Pull request merged**